internal/db/admin.go

```diff
@@ -0,0 +1,668 @@
+		return client.AddActorRelationshipResult{}, ErrAACRelationshipOperationRequiresIdentity
+	}
+
+	exists, err := db.adminInfo.AdminACP.AddActorRelationship(
+		ctx,
+		db.adminInfo.AdminDesc.Policy.Value().ID,
+		db.adminInfo.AdminDesc.Policy.Value().ResourceName,
+		acpTypes.InternalAdminACPObjectToGate,
+		relation,
+		requestActor.Value(),
+		targetActor,
+		protoTypes.TimestampNow(),
+	)
+	if err != nil {
+		return client.AddActorRelationshipResult{}, err
+	}
+
+	return client.AddActorRelationshipResult{ExistedAlready: exists}, nil
+}
+
+func (db *DB) deleteAACActorRelationship(
+	ctx context.Context,
+	relation string,
+	targetActor string,
+) (client.DeleteActorRelationshipResult, error) {
+	// Note: While we give unrestricted access when admin acp if turned off, there are certain
+	// requests that we can't do when admin acp is turned off or unavailable, this is one of them.
+	if !db.adminInfo.AdminDesc.IsEnabled || db.adminInfo.AdminACP == nil {
+		return client.DeleteActorRelationshipResult{}, client.ErrACPOperationButACPNotAvailable
+	}
+
+	if !db.adminInfo.AdminDesc.Policy.HasValue() {
+		return client.DeleteActorRelationshipResult{}, ErrAACIsEnabledButIsMissingPolicyInfo
+	}
+
+	policyDesc := db.adminInfo.AdminDesc.Policy.Value()
+	if policyDesc.ID == "" || policyDesc.ResourceName == "" {
+		return client.DeleteActorRelationshipResult{}, ErrAACIsEnabledButIsMissingPolicyInfo
+	}
+
+	requestActor := identity.FromContext(ctx)
+	if !requestActor.HasValue() || requestActor.Value() == nil || requestActor.Value().DID() == "" {
+		return client.DeleteActorRelationshipResult{}, ErrAACRelationshipOperationRequiresIdentity
+	}
+
+	recordFound, err := db.adminInfo.AdminACP.DeleteActorRelationship(
+		ctx,
+		policyDesc.ID,
+		policyDesc.ResourceName,
+		acpTypes.InternalAdminACPObjectToGate,
+		relation,
+		requestActor.Value(),
+		targetActor,
+		protoTypes.TimestampNow(),
+	)
+	if err != nil {
+		return client.DeleteActorRelationshipResult{}, err
+	}
+
+	return client.DeleteActorRelationshipResult{RecordFound: recordFound}, nil
+}
+
+func (db *DB) AddAACActorRelationship(
+	ctx context.Context,
+	relation string,
+	targetActor string,
+) (client.AddActorRelationshipResult, error) {
+	ctx, span := tracer.Start(ctx)
+	defer span.End()
+
+	if err := db.checkAdminAccess(ctx, acpTypes.AdminAACRelationAddPerm); err != nil {
+		return client.AddActorRelationshipResult{}, err
+	}
+
+	return db.addAACActorRelationship(ctx, relation, targetActor)
+}
+
+func (db *DB) DeleteAACActorRelationship(
+	ctx context.Context,
+	relation string,
+	targetActor string,
+) (client.DeleteActorRelationshipResult, error) {
+	ctx, span := tracer.Start(ctx)
+	defer span.End()
+
+	if err := db.checkAdminAccess(ctx, acpTypes.AdminAACRelationDeletePerm); err != nil {
+		return client.DeleteActorRelationshipResult{}, err
+	}
+
+	return db.deleteAACActorRelationship(ctx, relation, targetActor)
+}
```

Comment:
todo: These are public wrappers for the private methods. Please put each of them above the corresponding private methods so that it flows nicely for readers.